pkg/controller/factory/base_controller.go

```diff
@@ -216,29 +217,35 @@ func (c *baseController) degradedPanicHandler(panicVal interface{}) {
 	_ = c.reportDegraded(context.TODO(), fmt.Errorf("panic caught:\n%v", panicVal))
 }
 
+func (c *baseController) ControllerFieldManager(usageName string) string {
+	return ControllerFieldManager(c.name, usageName)
+}
+
 // reportDegraded updates status with an indication of degraded-ness
 func (c *baseController) reportDegraded(ctx context.Context, reportedError error) error {
 	if c.syncDegradedClient == nil {
 		return reportedError
 	}
 	if reportedError != nil {
-		_, _, updateErr := v1helpers.UpdateStatus(ctx, c.syncDegradedClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
-			Type:    c.name + "Degraded",
-			Status:  operatorv1.ConditionTrue,
-			Reason:  "SyncError",
-			Message: reportedError.Error(),
-		}))
+		condition := applyoperatorv1.OperatorStatus().
+			WithConditions(applyoperatorv1.OperatorCondition().
+				WithType(c.name + "Degraded").
+				WithStatus(operatorv1.ConditionTrue).
+				WithReason("SyncError").
+				WithMessage(reportedError.Error()))
+		updateErr := c.syncDegradedClient.ApplyOperatorStatus(ctx, c.ControllerFieldManager("reportDegraded"), condition)
 		if updateErr != nil {
 			klog.Warningf("Updating status of %q failed: %v", c.Name(), updateErr)
 		}
 		return reportedError
 	}
-	_, _, updateErr := v1helpers.UpdateStatus(ctx, c.syncDegradedClient,
-		v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
-			Type:   c.name + "Degraded",
-			Status: operatorv1.ConditionFalse,
-			Reason: "AsExpected",
-		}))
+
+	condition := applyoperatorv1.OperatorStatus().
+		WithConditions(applyoperatorv1.OperatorCondition().
+			WithType(c.name + "Degraded").
+			WithStatus(operatorv1.ConditionFalse).
+			WithReason("AsExpected"))
+	updateErr := c.syncDegradedClient.ApplyOperatorStatus(ctx, c.ControllerFieldManager("reportDegraded"), condition)
```

Comment:
> nit: up above we log a warning if `updateErr != nil`, should we do the same here or remove that one?

We do it up above because the update error is not reported.  Here we report the update failure.